services/payments/qldb_integration.go

```diff
@@ -0,0 +1,164 @@
+	Execute(ctx context.Context, fn func(txn qldbdriver.Transaction) (interface{}, error)) (interface{}, error)
+	Shutdown(ctx context.Context)
+}
+
+// WrappedQldbTxnAPI defines the API for QLDB methods that we'll be using
+type WrappedQldbTxnAPI interface {
+	Execute(statement string, parameters ...interface{}) (WrappedQldbResult, error)
+	Abort() error
+	BufferResult(*qldbdriver.Result) (*qldbdriver.BufferedResult, error)
+}
+
+// WrappedQldbResult defines the Result characteristics for QLDB methods that we'll be using
+type WrappedQldbResult interface {
+	Next(WrappedQldbTxnAPI) bool
+	GetCurrentData() []byte
+}
+
+// QLDBPaymentTransitionHistoryEntryBlockAddress defines blockAddress data for QLDBPaymentTransitionHistoryEntry
+type QLDBPaymentTransitionHistoryEntryBlockAddress struct {
+	StrandID   string `ion:"strandID"`
+	SequenceNo int64  `ion:"sequenceNo"`
+}
+
+// QLDBPaymentTransitionHistoryEntryHash defines hash for QLDBPaymentTransitionHistoryEntry
+type QLDBPaymentTransitionHistoryEntryHash string
+
+// QLDBPaymentTransitionHistoryEntrySignature defines signature for QLDBPaymentTransitionHistoryEntry
+type QLDBPaymentTransitionHistoryEntrySignature []byte
+
+// QLDBPaymentTransitionHistoryEntryData defines data for QLDBPaymentTransitionHistoryEntry
+type QLDBPaymentTransitionHistoryEntryData struct {
+	Status    QLDBPaymentTransitionState `ion:"status"`
+	Signature []byte                     `ion:"signature"`
+}
+
+// QLDBPaymentTransitionHistoryEntryMetadata defines metadata for QLDBPaymentTransitionHistoryEntry
+type QLDBPaymentTransitionHistoryEntryMetadata struct {
+	ID      string    `ion:"id"`
+	Version int64     `ion:"version"`
+	TxTime  time.Time `ion:"txTime"`
+	TxID    string    `ion:"txId"`
+}
+
+// QLDBPaymentTransitionHistoryEntry defines top level entry for a QLDB transaction
+type QLDBPaymentTransitionHistoryEntry struct {
+	BlockAddress QLDBPaymentTransitionHistoryEntryBlockAddress `ion:"blockAddress"`
+	Hash         QLDBPaymentTransitionHistoryEntryHash         `ion:"hash"`
+	Data         QLDBPaymentTransitionHistoryEntryData         `ion:"data"`
+	Metadata     QLDBPaymentTransitionHistoryEntryMetadata     `ion:"metadata"`
+}
+
+// BuildSigningBytes returns the bytes that should be signed over when creating a signature
+// for a QLDBPaymentTransitionHistoryEntry.
+func (e QLDBPaymentTransitionHistoryEntry) BuildSigningBytes() []byte {
+	return []byte(fmt.Sprintf("%s|%d|%s|%d|%s|%d|%s|%s",
+		e.BlockAddress.StrandID,
+		e.BlockAddress.SequenceNo,
+		e.Hash,
+		e.Data.Status,
+		e.Metadata.ID,
+		e.Metadata.Version,
+		e.Metadata.ID,
+		e.Metadata.TxID,
+	))
+}
+
+// GetTransitionHistory returns a slice of entries representing the entire state history
+// for a given id.
+func GetTransitionHistory(txn WrappedQldbTxnAPI, id string) ([]QLDBPaymentTransitionHistoryEntry, error) {
+	result, err := txn.Execute("SELECT * FROM history(PaymentTransitions) AS h WHERE h.metadata.id = 'SOME_ID'")
```

Comment:
Definitely the right thing and I'll go ahead and fix it. I was less specific with these SQL queries because I don't _think_ they're security relevant and they'll be much more specific as the tests are adapted to a real implementation.